api/src/main/java/org/openmrs/module/kenyaemr/calculation/library/mchms/DualHIVSyphilisCalculation.java

```diff
@@ -0,0 +1,140 @@
+            Obs pregStatusObs = EmrCalculationUtils.obsResultForPatient(pregStatusObss, ptId);
+
+            Obs hepBObs = EmrCalculationUtils.obsResultForPatient(lastHepatitisTesting, ptId);
+            Obs syphilsObs = EmrCalculationUtils.obsResultForPatient(lastSyphilisTesting, ptId);
+            Obs resultOfHivTestingObs = EmrCalculationUtils.obsResultForPatient(resultOfHivTesting, ptId);
+            Obs lmpObs = EmrCalculationUtils.obsResultForPatient(lmp, ptId);
+
+            lmpDate = lmpObs.getObsDatetime();
+
+            if(hepBObs !=null){
+                if (hepBObs.getObsDatetime() !=null){
+                    hepatitisTestDate = hepBObs.getObsDatetime();
+                    hepBTestMonths = monthsBetween(hepatitisTestDate, lmpDate);
+                }
+            }
+            if (syphilsObs != null) {
+                if (syphilsObs.getObsDatetime() != null) {
+                    syphilisTestDate = syphilsObs.getObsDatetime();
+                    syphilisTestMonths = monthsBetween(syphilisTestDate, lmpDate);
+                }
+            }
+
+            if(resultOfHivTestingObs !=null) {
+                if (resultOfHivTestingObs.getObsDatetime() != null) {
+                    hivTestDate = resultOfHivTestingObs.getObsDatetime();
+                    hivTestMonths = monthsBetween(hivTestDate, lmpDate);
+
+                }
+            }
+
+            int pregnancyMonths = monthsBetween(currentDate, lmpDate);
+            int monthsSinceFirstTrimester = monthsBetween(lmpDate, currentDate);
+            boolean syphilisQuestion = lastANCEnc != null ? EmrUtils.encounterThatPassCodedAnswer(lastANCEnc, syphilisQ, syphilisNegative) : false;
+
+            if(pregStatusObs != null && pregStatusObs.getValueCoded().equals(yes)){
+               if(lastANCEnc !=null) {
+
+                   if (pregnancyMonths >= 1 && pregnancyMonths <= 3) {
+                       if (hepBObs ==null && (hepBTestMonths < 1 || hepBTestMonths > 3)) {
+                           eligibleDualHivSyphilisFlag = true;
+                        dualMessage = "Due for HepB";
+                       }
+                       if (syphilsObs ==null && (syphilisTestMonths < 1 || syphilisTestMonths > 3)) {
+                           eligibleDualHivSyphilisFlag = true;
+                           dualMessage = "Due for Syphils";
+                       }
+                       if (resultOfHivTestingObs ==null && (hivTestMonths < 1 || hivTestMonths > 3)) {
+                           eligibleDualHivSyphilisFlag = true;
+                           dualMessage = "Due for Hiv Test";
+                       }
+                   }
+                if (pregnancyMonths >= 7 && pregnancyMonths <= 9){
+                       if(resultOfHivTestingObs != null && syphilisQuestion &&  resultOfHivTestingObs.getValueCoded().equals(negative)  && monthsSinceFirstTrimester > 3) {
```

Comment:
1. What if The client **never** had either Syphillis or HIV tests or Both tests before? 
2. Do you really need **monthsSinceFirstTrimester > 3** in this condition?
3. What if the client had one test and not the other within the 3rd trimester?
4. What if the client had both tests in the third trimester as expected? Will this flag disappear?

Reply:
All the four condtions fixed